File: Model.Tests/MatchGenerators/MatchDaysMatchGeneratorTests.vb
<TestClass>
Public Class MatchDaysMatchGeneratorTests

    <TestMethod>
    Public Sub Generate2Test()
        Dim sut = New MatchdaysMatchGenerator(2)
        Dim result = sut.Generate()
        Assert.AreEqual(1, result.Count)
    End Sub

    <TestMethod>
    Public Sub Generate4Test()
        Dim sut = New MatchdaysMatchGenerator(4)
        Dim result = sut.Generate()
        Assert.AreEqual(6, result.Count)
    End Sub

    <TestMethod>
    Public Sub Generate10Test()
        Dim sut = New MatchdaysMatchGenerator(10)
        Dim result = sut.Generate()
        Assert.AreEqual(45, result.Count)
        Assert.AreNotEqual(New Tuple(Of Integer, Integer)(0, 2), result.ElementAt(1))
    End Sub

    <TestMethod>
    Public Sub GenerateXTest()
        Dim sut = New MatchdaysMatchGenerator(14)
        Dim result = sut.Generate()
        For Each matchIndexPair In result
            Console.WriteLine(matchIndexPair)
        Next
    End Sub

End Class

File: POFF.Kicker.Tests/MatchDaysMatchGeneratorTests.cs
using System;
using System.Linq;
using NUnit.Framework;
using POFF.Kicker.Domain.MatchGenerators;

namespace POFF.Kicker.Tests;

[TestFixture]
class MatchDaysMatchGeneratorTests
{
    [Test]
    public void Generate2Test()
    {
        var sut = new MatchdaysMatchGenerator(2);
        var result = sut.Generate();
        Assert.That(result.Count(), Is.EqualTo(1));
    }

    [Test]
    public void Generate4Test()
    {
        var sut = new MatchdaysMatchGenerator(4);
        var result = sut.Generate();
        Assert.That(result.Count(), Is.EqualTo(6));
    }

    [Test]
    public void Generate10Test()
    {
        var sut = new MatchdaysMatchGenerator(10);
        var result = sut.Generate();
        Assert.That(result.Count(), Is.EqualTo(45));
        Assert.AreNotEqual(new Tuple<int, int>(0, 2), result.ElementAt(1));
    }

    [Test]
    public void GenerateXTest()
    {
        var sut = new MatchdaysMatchGenerator(14);
        var result = sut.Generate();
        foreach (var matchIndexPair in result)
            Console.WriteLine(matchIndexPair);
    }
}